### PR TITLE
adding health functionality and adding more metric

### PR DESCRIPTION
**Pull request by vsacheti**

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: lib/index.js [hunk 1]
```diff
@@ -199,6 +218,35 @@ Watcher.prototype = {
...
+                }
+            }
+        }
+        for (i in statuses) {
+            if (statuses[i].oreqs) {
+                totalOpenRequests += statuses[i].oreqs;
+            }
+        }
+        wmetric.set({ "watcher.proc.openreqs" : totalOpenRequests }, 0);
+        if (!process.yahoo) {
```
Comment (rohiniwork) [inline lib/index.js]: You will have to remove all this stuff or rename appropriately

**Comment by vsacheti** [comment]
Rohini, updated with your comments. Not sure if you can still review it :), but if you can that will be great.

**Comment by rohiniwork** [comment]
I can surely do that :)

File: lib/index.js [hunk 2]
```diff
@@ -276,9 +324,28 @@ Watcher.prototype = {
...
                 wmetric.increment("watcher.proc.killed");
                 delete statuses[i];
+                this.postDelete(i);
             }
         }
+    },
+    postDelete : function (pid) {
+        //if health is up already, nothing to do
+        if (this.healthChangeCallback && this.overall_health_is_down && !this.isAnyWorkerDown()) {
+            //health is down, but none of the workers are down
```
Comment (rohiniwork) [inline lib/index.js:338]: I do not have any other comment but just this one.

Why is the toggling tied to the presence of a callback? Either way u will need to toggle back the health to UP if the bad one is already killed.
Hope I havent forgotten anything we already discussed before :)

**Comment by vsacheti** [comment]
Rohini, Good point. I have updated the PR.

**Comment by rohiniwork** [comment]
I think u can go ahead with this.
I dont see any other issue. May be Evan could check once?

**Pull request merged**